splunk_connect_for_snmp_poller/manager/hec_sender.py

```diff
@@ -77,12 +92,42 @@ def post_event_data(endpoint, host, variables_binds, index, one_time_flag=False)
         logger.error(f"Connection error when sending data to HEC index - {index}: {e}")
 
 
-def post_metric_data(endpoint, host, variables_binds, index):
-
+def _enrich_event_data(mib_enricher: MibEnricher, variables_binds: dict) -> str:
+    """
+    This function serves for processing event data the way we add additional dimensions configured in enricher config.
+    @param mib_enricher: MibEnricher object containing additional dimensions
+    @param variables_binds: dictionary containing "metric_name", "metric" - metric version of varbinds and
+    "non_metric" - nonmetric version of varbinds, for ex:
+
+    {'metric': '{"metric_name": "sc4snmp.IF-MIB.ifPhysAddress_1", "_value": "", "metric_type": "OctetString"}',
+    'metric_name': 'sc4snmp.IF-MIB.ifPhysAddress_1',
+    'non_metric': 'oid-type1="ObjectIdentity" value1-type="OctetString" 1.3.6.1.2.1.2.2.1.6.1=""
+    value1="" IF-MIB::ifPhysAddress.1="" '}
+
+    We need both formats because process_one function was designed to work on metric data only and non metric format is
+    difficult to process because of the nature of string type.
+
+    @return: non metric varbind with values from additional dimension added. For ex. for additional dimensions:
+    [interface_index, interface_desc]:
+    'oid-type1="ObjectIdentity" value1-type="OctetString" 1.3.6.1.2.1.2.2.1.6.1="" value1="" IF-MIB::ifPhysAddress.1=""
+    interface_index="1" interface_desc="lo" '
+    """
+    metric_result = json.loads(variables_binds["metric"])
+    non_metric_result = variables_binds["non_metric"]
+    mib_enricher.process_one(metric_result)
+    for field_name in mib_enricher.dimensions_fields:
+        if field_name in metric_result:
+            non_metric_result += f'{field_name}="{metric_result[field_name]}" '
```

Comment:
If we want to have field names separated by space we append it at the end of each iteration and rely on variables_binds["non_metric"] to have trailing space (most probbably added by MIB server). When there are no dimensions_fields we will have trailing space on the output. Not sure if this is a problem though.

Reply:
This is the default way text data is being returned from mib server. I'm not sure if it's correct, but after this PR nothing will be changed.